containerization/docker/fedora/Dockerfile

```diff
@@ -0,0 +1,41 @@
+# SPDX-License-Identifier: BSD-3-Clause
```

Comment:
Something we should do in a future PR is to create a build matrix for the "docker" workflow (.github/workflows/docker.yml) so we'll know if the container build is broken for Fedora. It also has the added benefit of testing compilation on Fedora.